llvm/include/llvm/CodeGen/BasicTTIImpl.h

```diff
@@ -991,17 +991,34 @@ class BasicTTIImplBase : public TargetTransformInfoImplCRTPBase<T> {
       return Kind;
     int NumSrcElts = Ty->getElementCount().getKnownMinValue();
     switch (Kind) {
-    case TTI::SK_PermuteSingleSrc:
+    case TTI::SK_PermuteSingleSrc: {
       if (ShuffleVectorInst::isReverseMask(Mask, NumSrcElts))
         return TTI::SK_Reverse;
       if (ShuffleVectorInst::isZeroEltSplatMask(Mask, NumSrcElts))
         return TTI::SK_Broadcast;
+      // Check that the broadcast index meets at least twice.
+      bool IsCompared = false;
```

Comment:
Would be easier to read if you pulled out a helper function for this, and left a comment about why this is different than isZeroEltSplatMask

Reply:
You mean, add a new helper in ShuffleVectorInst?

Reply:
Ideally, but even just local to the file would help.